marketplace_tracker.js

```diff
@@ -70,6 +71,31 @@ function getSellPrice() {
     }
 }
 
+function favoriteButton(currentItem){
+    if(document.getElementById("marketplace-refresh-button").nextSibling==document.getElementById("marketplace-item-list-searchbar")){
+        //above: refresh doesn't add buttons
+        sendMessage({
+            type: 'get-favorite',
+            item: currentItem
+        }).then(response => {
+            console.log(response);
+            document.getElementById("marketplace-refresh-button").insertAdjacentHTML("afterend", favoriteTemplate(response.isFavorite));
+            let favoriteButton = document.getElementById("marketplace-favorite-button")
+            favoriteButton.addEventListener('click', function(){
+                sendMessage({
+                    type: 'toggle-favorite',
+                    item: currentItem
+                }).then(response => {
+                    console.log(response.success);
+                    console.log(response.isFavorite);
```

Comment:
A lot of leftover `console.log`s in the code.
Btw you can just log response directly here. Less console output